### PR TITLE
gdlv: init at 1.8.0

### PR DESCRIPTION
**Pull request by mmlb**
###### Description of changes

gdlv is a graphical frontend to Delve. Delve is usually used with editor integration, but gdlv allows editor independent usage.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/gdlv/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "gdlv";
+  version = "1.8.0";
+
+  src = fetchFromGitHub {
+    owner = "aarzilli";
+    repo = "${pname}";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/gdlv/default.nix:9]:
```suggestion
    repo = pname;
```

or better

```suggestion
    repo = "gdlv";
```

Comment (mmlb) [reply pkgs/development/tools/gdlv/default.nix]: going with the latter, thanks for the review!

**Comment by mmlb** [comment]
@SuperSandro2000 PTAnotherL

File: pkgs/development/tools/gdlv/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{ lib, buildGoModule, fetchFromGitHub }:
```
Comment (SuperSandro2000) [inline pkgs/development/tools/gdlv/default.nix:1]:
```suggestion
{ lib, buildGoModule, fetchFromGitHub, CoreFoundation }:
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -24844,6 +24844,8 @@ with pkgs;
 
   ginkgo = callPackage ../development/tools/ginkgo { };
 
+  gdlv = callPackage ../development/tools/gdlv { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:24847]:
```suggestion
  gdlv = callPackage ../development/tools/gdlv {
    inherit (darwin.apple_sdk.frameworks) CoreFoundation;
  };
```

**Comment by SuperSandro2000** [comment]
@ofborg build gdlv

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**